src/symbolic_dimensions.jl

```diff
@@ -34,31 +40,47 @@ You can convert a quantity using `SymbolicDimensions` as its dimensions
 to one which uses `Dimensions` as its dimensions (i.e., base SI units)
 `uexpand`.
 """
-struct SymbolicDimensions{R} <: AbstractDimensions{R}
+struct SymbolicDimensions{R} <: AbstractSymbolicDimensions{R}
     nzdims::Vector{INDEX_TYPE}
     nzvals::Vector{R}
 end
 
-@inline dimension_names(::Type{<:SymbolicDimensions}) = ALL_SYMBOLS
+"""
+    SymbolicDimensionsSingleton{R} <: AbstractSymbolicDimensions{R}
+
+This special symbolic dimensions types stores a single unit or constant, and can
+be used for constructing symbolic units and constants without needing to allocate mutable storage.
```

Comment:
Given also the comments below about allocations in `nzdims` etc. I wonder if it would be easier to allow non-mutable containers (`Tuple`s?) in `SymbolicDimensions`.

Reply:
I'm afraid this would be a lot more complexity than it would save as you would have to account for additional type parameters in `AbstractDimensions` types which I haven't tested yet. But it's only a single edge case for right now. Maybe if the need arises we could take a look.

Reply:
Actually, wait... This could actually be very cool, because (and maybe this is why you suggested it?), this would technically be a way to have full static-typing equivalence to Unitful.